browser-extension/src/models/banner/Banner.js

```diff
@@ -0,0 +1,156 @@
+import { isNavigatorLang } from '../../utils.js';
+import { defaultBannerOptions } from './BannerOptions.js';
+
+export class Banner {
+  constructor(options = defaultBannerOptions) {
+    this.TS_BANNER_DATE_START = options.TS_BANNER_DATE_START;
+    this.TS_BANNER_DATE_END = options.TS_BANNER_DATE_END;
+    this.TS_BANNER_ACTION_FR_URL = options.TS_BANNER_ACTION_FR_URL;
+    this.TS_BANNER_ACTION_EN_URL = options.TS_BANNER_ACTION_EN_URL;
+    this.TS_BANNER_PROOF_KW = options.TS_BANNER_PROOF_KW;
+
+    // Create the banner once at the initialisation
+    this.banner = this.createBanner();
+  }
+
+  displayBanner() {
```

Comment:
I think there is no need to use the `Banner` suffix here, as we are already in a banner.

This change implies the update of all calls to  `displayBanner`.

```suggestion
  display() {
```